### PR TITLE
Include team reads and unreads

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
Handle messages to mark the team as read or unread

#### Ticket Link
 https://mattermost.atlassian.net/browse/RN-137

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Added or updated unit tests (required for all new features)
- [x] All new/modified APIs include changes to the drivers

### CODE REVIEW

File: src/actions/channels.js [hunk 1]
```diff
@@ -793,14 +793,22 @@ export function updateChannelPurpose(channelId, purpose) {
...
-        if (channels[channelId] && channelMember) {
-            totalMsgCount = channels[channelId].total_msg_count;
+        let teamMember;
+        if (channel && channelMember) {
+            totalMsgCount = channel.total_msg_count;
+
+            if (channel.team_id) {
+                teamMember = {...teamMembers[channel.team_id]};
+                teamMember.mention_count = teamMember.mention_count - channelMember.mention_count;
+                teamMember.msg_count = teamMember.msg_count - (totalMsgCount ? (totalMsgCount - channelMember.msg_count) : totalMsgCount);
```
Comment (jwilander) [inline src/actions/channels.js]: This line is a bit confusing to read but not a big deal

File: src/actions/channels.js [hunk 2]
```diff
@@ -827,17 +843,34 @@ export function markChannelAsRead(channelId, prevChannelId) {
...
+        }
+
         if (actions.length) {
             dispatch(batchActions([...actions]), getState);
         }
     };
 }
 
-export function markChannelAsUnread(channelId, mentionsArray) {
+export function markChannelAsUnread(teamId, channelId, mentionsArray) {
```
Comment (jwilander) [inline src/actions/channels.js]: Is it possible to get the team id from the channel object so we don't need to pass it in here? If it's too much work then this is fine, 0/5

File: src/actions/users.js [hunk 3]
```diff
@@ -151,11 +158,13 @@ export function loadMe() {
         const prefs = getMyPreferences()(dispatch, getState);
         const teams = getMyTeams()(dispatch, getState);
         const members = getMyTeamMembers()(dispatch, getState);
+        const teamUnreads = getMyTeamUnreads()(dispatch, getState);
 
         await me;
         await prefs;
         await teams;
         await members;
+        await teamUnreads;
```
Comment (jwilander) [inline src/actions/users.js]: Do we need to wait on this or is it ok if it pops in a short time after loading? Just wondering if we can remove the `await`

File: src/actions/channels.js (hunk 2 above)
Comment (enahum) [inline src/actions/channels.js]: The problem with that is you may get a post from a channel in a team that hasn't been loaded yet and then you don't have the channel in the store, no channel in the store means no team_id, sounds about right?

File: src/actions/users.js (hunk 3 above)
Comment (enahum) [inline src/actions/users.js]: can do

**Comment by enahum** [comment]
@jwilander done

**Pull request merged**